cmd/flags.go

```diff
@@ -61,11 +61,11 @@ func FlagBuildpacksDir(dir *string) {
 }
 
 func FlagCacheDir(dir *string) {
-	flag.StringVar(dir, "path", os.Getenv(EnvCacheDir), "path to cache directory")
+	flag.StringVar(dir, "cache-dir", os.Getenv(EnvCacheDir), "path to cache directory")
```

Comment:
could we accept `-path` or `-cache-dir` to prevent a hard break?

Reply:
`cacher` is going away entirely, so this is going to be a hard break for everyone anyways.

Unrelated comment: In the RFC we discussed this being `-cache-volume`, and it seems to have been changed to `-cache-dir` here. (Perhaps this is because it doesn't need to be a volume?) It might be worth pointing out that for the rest of the flags, we don't append `-dir` or `-path` the argument takes a path. Would `-cache` be more appropriate?

Reply:
I chose cache-dir to match the existing CNB_CACHE_DIR environment variable. If we want to go with -cache-volume instead, we should probably update the env var to match. While it doesn't necessarily need to be a volume, I think it will likely always be as the cache would otherwise effectively get discarded after each build. That being said if we're okay with changing the env var, I don't have a strong preference for either -cache-dir or -cache-volume.

I slightly prefer longer '-cache-dir' as opposed to just '-cache' as it's matches other other cache flag, '-cache-image' as well as the env var above. But if you'd like to maintain flag consistency, I'm not opposed to caching it.

Reply:
@sclevine `-path` is on `restorer`, which is not going away?

Instead of immediately `rm -rf`ing `cacher` in the next release, we could leave the binary in place as a no-op and have it emit a warning that it has been deprecated and will be removed in a future release. I think we're at a point where we should consider this, especially if it's the only breaking change in a given release.

Reply:
@jkutner If a platform is unaware of the platform API change, having `cacher` no-op here might create unexpected behavior.

If we wanted to prevent a hard break, these are the options as I see them:
1. We have `exporter` skip the caching step when it is provided neither a dir nor an image. This will be slow but won't implode. It has some weird edge cases given that `restorer` will still restore the last successful cache.
2. We could give `exporter` default caching behavior that matches `pack`'s current usage (i.e cache to `/cache`). This works for `pack` but would do the wrong thing and or break for a platform that is currently using `cacher` with `-image` for example

I don't particularly like either of those options.

Given that an increase in the Platform API to `0.2` indicates a breaking change, I think we could just trust platforms to know if they aren't compatible with this lifecycle rather than trying to prevent a breaking change (`pack` will already bail out in this case).

Reply:
I was thinking the same thing as @jkutner. I don't think it would be difficult to provide an intermediate shim. This would prevent unexpectedly breaking from one builder to the next. This would affect all _platform_ implementations like `pack` as well as tekton, and knative definitions. Allowing for some grace period would probably be beneficial by preventing unnecessary support (via slack, issues, or otherwise).

Reply:
In that case, I think we would need to leave an actual implementation of `cacher` , not a no-op binary, b/c those platforms won't be providing the necessary cache flags to `exporter`.

Reply:
Yea i prefer leaving the `cacher` option, which is what i already suggested in https://github.com/buildpack/lifecycle/pull/205#issuecomment-546950819 (not sure why switched :P). Also still open to making a breaking change, but it feels like we don't *have* to.

Reply:
> @sclevine -path is on restorer, which is not going away?
The flag name affects restorer as well, since it also uses FlagCacheDir.

If we leave the cacher binary as is, what happens in the case where the platform sets flags through env vars? Platforms aware of this change can remove the cacher step, but for those that keep it would both exporter and cacher write to the same cache image twice?

Part of the RFC is switching the order of analyzer and restorer (in progress as a separate PR), which significantly alters their behavior and may necessitate a breaking change anyway. (At least I'm not sure I see a way to implement the changes in a backwards-compatible way.)

Reply:
Discussed offline, we will resolve this by updating the platform api version and releasing an updating version of pack.

Reply:
Sounds good, thanks @lukasberger. I'd like to keep track of all the changes so that we can write a migration guide for 0.6.0 (or whatever version this lands in). I think the updates to `pack` will help us catalog those changes.

Reply:
Please see https://github.com/buildpack/pack/pull/384 for the proposed pack changes.

Reply:
Sorry for the nitpick, specially if the conversation was had and I just couldn't find references to it. I'm in favor for the name `cache-dir` as oppose to `cache-volume` since technically, depending on the system, it might not actually be a volume, it's really just a directory.

My nitpick is around consistency, we have other directories that don't have the `dir` suffix. For example, `layers` , `platform`, `launch-cache`. Same for `image` suffix. For consistencies sake and to prevent yet another breaking change to rename them in the future what do we think about renaming them all at once?

Proposal:

- `layers` -> `layers-dir`
- `platform` -> `platform-dir`
- `launch-cache` -> `launch-cache-dir`
- `image` -> `run-image`

IMO, they are more understandable with the slight addition of a suffix.

Reply:
@jromero modifying all of these flag names would be a big change require spec changes and probably deserves it's own RFC. I think we should discuss that idea separately from this PR.